Sources/MapboxCoreNavigation/NavigationService.swift

```diff
@@ -266,7 +266,9 @@ public class MapboxNavigationService: NSObject, NavigationService {
     
     deinit {
         suspendNotifications()
-        endNavigation()
+        eventsManager.withBackupDataSource(self) {
+            endNavigation()
```

Comment:
NIT: A potential issue might be if there are async operations inside

Reply:
True. But I don't see any better options